miniml/compiler/compile.scm

```diff
@@ -529,6 +550,30 @@
           (else #nil)
           )))
 
+(define (symbol-char? c)
+  (member c (list #\! #\$ #\% #\& #\* #\+ #\- #\. #\/ #\: #\< #\= #\> #\? #\@ #\^ #\| #\~) char=?))
+
+(define (symbol-chars errorp)
+  (let ((c (peek-char)))
+    (cond ((eof-object? c) #nil)
+          ((symbol-char? c) (begin (read-char) (cons c (symbol-chars errorp))))
+          (else #nil)
+          )))
+
+(define (get-operator name)
+  (let ((p (assoc name operator-kw))
+        (c (string-ref name 0)))
+    (cond (p (cdr p))
+          ((member c (list #\! #\~ #\?) char=?) (cons 'PREFIXOP name))
```

Comment:
(here again string literals could make this nicer to read, although it's less of an issue as the lists are much shorter)